test/helper.rb

```diff
@@ -649,3 +649,13 @@ def load_resource(*args)
   m = MockLoader.new(:ubuntu1404)
   m.send('load_resource', *args)
 end
+
+# Used to capture `Inspec.deprecate()` with warn action
+def expect_deprecation_warning
```

Comment:
We might want to make this looser - like `expect_deprecation_action(opts)`.  With no args, it expects that `Inspec.deprecate` will be called at least once.  With args, we might insist on a particular action, or restrict which group names we expect to see.

The goal here would be that a deprecation will naturally progress through ignore, warn, fail/exit over its life, letting us change the test code less frequently.  If we use a more general check like this, we would not be insisting on a warning, just a check that we know it is scheduled for deprecation.

@miah, thoughts?